core/builtins/src/kotlin/Progressions.kt

```diff
@@ -110,6 +122,18 @@ public open class IntProgression
 
     override fun toString(): String = if (step > 0) "$first..$last step $step" else "$first downTo $last step ${-step}"
 
+    override val size: Int
+        get() = if (isEmpty()) 0 else (last - first) / step + 1
+
+    override fun contains(@Suppress("PARAMETER_NAME_CHANGED_ON_OVERRIDE") /* for the backward compatibility with old names */ value: Int): Boolean = when {
+        this.isEmpty() -> false
+        step > 0 && value >= first && value <= last -> (value - first) % step == 0
```

Comment:
I wonder if `(value - first) % step` is overflow safe?
For example, `2147483649 % 3 == 0`, but the corresponding overflown value `-2147483647 % 3 == -1`

Reply:
I will replace it with `abs(value % step) == abs(first % step)` and will add some tests for overflow.

Reply:
I don't think it will do. Consider for example, a progression with `first = -1` and `step = 4`: the `value = 3` should be in it, but it has a different remainder than -1 has.

Reply:
As I haven't run the tests yet, I haven't found that. Thank you! (I still haven't run them, but I think that using `mod` from [ticket](https://youtrack.jetbrains.com/issue/KT-26234) would help).